internal/pkg/aws/cloudformation/stackset/errors.go

```diff
@@ -13,12 +14,50 @@ import (
 // ErrStackSetOutOfDate occurs when we try to read and then update a StackSet but between reading it
 // and actually updating it, someone else either started or completed an update.
 type ErrStackSetOutOfDate struct {
-	stackSetName string
-	parentErr    error
+	name      string
+	parentErr error
 }
 
 func (e *ErrStackSetOutOfDate) Error() string {
-	return fmt.Sprintf("stack set %s update was out of date (feel free to try again): %v", e.stackSetName, e.parentErr)
+	return fmt.Sprintf("stack set %q update was out of date (feel free to try again): %v", e.name, e.parentErr)
+}
+
+// ErrStackSetNotFound occurs when a stack set with the given name does not exist.
+type ErrStackSetNotFound struct {
+	name string
+}
+
+// Error implements the error interface.
+func (e *ErrStackSetNotFound) Error() string {
+	return fmt.Sprintf("stack set %q not found", e.name)
+}
+
+func (e *ErrStackSetNotFound) isEmpty() bool {
+	return true
+}
+
+// ErrStackSetInstancesNotFound occurs when a stack set operation should be applied to instances but they don't exist.
+type ErrStackSetInstancesNotFound struct {
+	name string
+}
+
+// Error implements the error interface.
+func (e *ErrStackSetInstancesNotFound) Error() string {
+	return fmt.Sprintf("stack set %q has no instances", e.name)
+}
+
+func (e *ErrStackSetInstancesNotFound) isEmpty() bool {
+	return true
+}
+
+// IsEmptyStackSetErr returns true if the error occurred because the stack set does not exist or does not contain any instances.
+func IsEmptyStackSetErr(err error) bool {
```

Comment:
Would you consider this as a good alternative: defining `IsEmptyStackSetErr` in `cloudformation` package (which is the only place this seems to get used) and export `isEmpty`?

Reply:
That is fine too! I also realized that i should have named it `IsEmptyErr` so that it reads `stackset.IsEmptyErr` instead of stuttering.

I like your suggestion, if we're going to have both `stackset` and `aws/cloudformation` implement the `isEmpty` interface then it makes sense to move it to `deploy/cloudformation`.

Side note: this pattern is similar to how `os.IsTimeout` works in the stdlib:  https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/os/error.go;l=114

And there are errors like `SyscallError` that implements the internal `timeout` interface: https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/os/error.go;l=61

but also errors in the sub package of `fs`: https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/io/fs/fs.go;l=255

I'll updated the code to `cloudformation.IsEmptyErr` 👍